File: API/Models/UserRate.cs
namespace Brickalytics.Models
{
    public class UserRate
    {
        public int UserId {get; set;}
        public int? ProductTypeId {get; set;}
        public int ProductSubTypeId {get; set;}
        public decimal? Rate { get; set; }
        public decimal? Percent { get; set; }
    }
}

File: UI/src/app/core/models/userRate.ts
export class UserRate {
  userId: number;
  productTypeId: number;
  rate: number;
  percent: number;

  constructor(
    userId: number,
    productTypeId: number,
    rate: number,
    percent: number,
  ) {
    this.userId = userId;
    this.productTypeId = productTypeId;
    this.rate = rate;
    this.percent = percent;
  }
}
